subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/problems/InstantExecutionProblems.kt

```diff
@@ -56,27 +59,47 @@ class InstantExecutionProblems(
     private
     val buildFinishedHandler = BuildFinishedProblemsHandler()
 
+    private
+    val postBuildHandler = PostBuildProblemsHandler()
+
     private
     val problems = CopyOnWriteArrayList<PropertyProblem>()
 
     private
     var isFailOnProblems = startParameter.failOnProblems
 
+    private
+    var isFailingBuildDueToSerializationError = false
+
+    private
+    var isLoading = false
+
     init {
         listenerManager.addListener(problemHandler)
         listenerManager.addListener(buildFinishedHandler)
+        listenerManager.addListener(postBuildHandler)
     }
 
     override fun close() {
         listenerManager.removeListener(problemHandler)
         listenerManager.removeListener(buildFinishedHandler)
+        listenerManager.removeListener(postBuildHandler)
+    }
+
+    fun storing() {
+        isLoading = false
```

Comment:
❌ An enum with the two states, loading and storing, would be clearer.

Reply:
Done